test/evergreen/tsan_warnings_analysis.py

```diff
@@ -25,55 +25,93 @@
 # OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
 # ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 # OTHER DEALINGS IN THE SOFTWARE.
-import os, sys
-import re, subprocess
-# The script intends to generate warnings from WiredTiger's examples, csuite, python and test/format testing.
+import argparse, os, subprocess
+import re
+# The script will look through all tsan logs from the current directory
+# and collect all the warnings the found under tsan logs.
 #
-# Running the whole suite causes enormous number of TSAN warnings. The amount of I/O causes slowness in the
-# system leading to non-deterministic results. To ensure deterministic results run with only examples suite.
+# The TSAN configuration will need option log_path to create the tsan logs.
+def get_line_last_modified_times(file_path, line_number):
+    """
+    Get the last modification time (UNIX timestamp) of every line in a file using Git blame.
 
-# Configure log path in TSAN options.
-current_dir = os.getcwd()
-existing_tsan_options = os.environ.get("TSAN_OPTIONS")
-os.environ["TSAN_OPTIONS"] = f"{existing_tsan_options}:log_path={current_dir}/tsan_logs:exitcode=0"
+    :param file_path: Path to the file to analyze.
+    :param git_rev: Git revision (default: HEAD).
+    :return: List of tuples (line_number, line_content, last_modified_timestamp).
+    """
+    # Run `git blame` to retrieve metadata for found line number
+    git_command = ["git", "blame", "--line-porcelain", "HEAD", f"-L {line_number},{line_number}", "--", file_path]
+    result = subprocess.run(git_command, capture_output=True, text=True, check=True)
 
-# Start with examples suite.
-test_tasks = [
-    "ctest -j 8 --test-dir examples/c"
-]
+    timestamp = None
+    output_lines = result.stdout.splitlines()
+    for line in output_lines:
+        # Extract the UNIX timestamp
+        if line.startswith("author-time"):
+            timestamp = int(line.split()[1])
+    return timestamp
 
-# Run the tasks in list.
-print("Tasks to be executed")
-print(test_tasks)
-for task in test_tasks:
-    try:
-        # Tasks are allowed to run with maximum of three hours.
-        subprocess.run(task.split(), check=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, timeout=60 * 60 * 3)
-    except subprocess.CalledProcessError as exception:
-        print(f'Command {exception.cmd} failed with error {exception.returncode}')
-    except subprocess.TimeoutExpired as exception:
-        print(f'Command {exception.cmd} timed out')
-        sys.exit(1)
+def get_tsan_warnings():
+    """
+    Get the TSAN warnings from the log files.
+    :return: Set of unique TSAN warnings.
+    """
+    tsan_warnings_set = set()
```

Comment:
It seems like ["Set items can appear in a different order every time you use them, and cannot be referred to by index or key"](https://www.w3schools.com/python/python_sets.asp) so that could potentially lead to non-deterministic output on some platforms.

Reply:
IMO I think it is okay to have non-deterministic output, the unordered set is to make sure we have unique warnings. We don't need to index into the set via an index or key here.  

If we want it to be alphabetical we could also sort it before printing it out